src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -847,22 +818,18 @@ def load_module(name, value):
         # 7. Load each module in the pipeline
         current_device_map = None
         for name, (library_name, class_name) in logging.tqdm(init_dict.items(), desc="Loading pipeline components..."):
-            if final_device_map is not None and len(final_device_map) > 0:
-                component_device = final_device_map.get(name, None)
-                if component_device is not None:
-                    current_device_map = {"": component_device}
-                else:
-                    current_device_map = None
+            # 7.1 device_map shenanigans
+            current_device_map = _determine_current_device_map(device_map=final_device_map, component_name=name)
```

Comment:
here too,  I think it is better without the method

Reply:
Agreed. Done in d24465e04.